### PR TITLE
docs(filter): Add Filter Rebind() method

### PR DESCRIPTION
**Pull request by NansiYancheva**
https://github.com/telerik/blazor/issues/8902

### CODE REVIEW

File: components/filter/overview.md [hunk 1]
```diff
@@ -65,6 +65,123 @@ The Blazor Filter provides parameters that allow you to configure the component:
 | `Class` | `string` | The class that will be rendered on the outermost element. |
 | `Value` | `CompositeFilterDescriptor` | Sets the value of the filter component. |
 
+## Reference and Methods
```
Comment (dimodi) [inline components/filter/overview.md:68]:
```suggestion
## Filter Reference and Methods
```

File: components/filter/overview.md [hunk 2]
```diff
@@ -65,6 +65,123 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+
+The Filter exposes methods for programmatic operation. To use them, define a reference to the component instance with the `@ref` directive attribute.
+
+| Method | Description |
+| --- | --- |
+| `Rebind` | Calls `StateHasChanged` inside the Filter component. |
```
Comment (dimodi) [inline components/filter/overview.md:74]:
```suggestion
| `Rebind` | Processes the component `Value` and updates the component UI. |
```

File: components/filter/overview.md [hunk 3]
```diff
@@ -65,6 +65,123 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+
+>caption Get a reference to the Filter component and use its methods.
```
Comment (dimodi) [inline components/filter/overview.md:76]:
```suggestion
>caption Using the Filter component reference and methods
```

File: components/filter/overview.md [hunk 4]
```diff
@@ -65,6 +65,123 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+             Height="400px">
+    <GridColumns>
+        <GridColumn Field="@(nameof(Person.EmployeeId))" Title="Id" />
+        <GridColumn Field="@(nameof(Person.Name))" Title="First Name" />
+        <GridColumn Field="@(nameof(Person.AgeInYears))" Title="Age" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+
```
Comment (dimodi) [inline components/filter/overview.md:100]:
```suggestion
```

File: components/filter/overview.md [hunk 5]
```diff
@@ -65,6 +65,123 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+    private List<Person> GridData { get; set; } = new();
+
+    private List<Person> InitialData { get; set; } = new();
+
+    private TelerikFilter? FilterRef { get; set; }
```
Comment (dimodi) [inline components/filter/overview.md:105]: Define `@ref`s first

File: components/filter/overview.md [hunk 6]
```diff
@@ -65,6 +65,123 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+                    Operator = FilterOperator.IsEqualTo
+                },
+                new FilterDescriptor
+                {
+                    Member = nameof(Person.AgeInYears),
+                    MemberType = typeof(int),
+                    Operator = FilterOperator.IsEqualTo
+                },
+            };
+            FilterRef.Rebind();
```
Comment (dimodi) [inline components/filter/overview.md:150]:
```suggestion
            await Task.Delay(1000);
            FilterRef.Rebind();
```

File: components/filter/overview.md [hunk 7]
```diff
@@ -65,6 +65,123 @@ The Blazor Filter provides parameters that allow you to configure the component:
...
+        GridData = dataSourceResult.Data.Cast<Person>().ToList();
+    }
+
+    protected override async Task OnAfterRenderAsync(bool firstRender)
+    {
+        if (firstRender)
+        {
+            FilterValue.FilterDescriptors.Clear();
+            FilterValue.LogicalOperator = FilterCompositionLogicalOperator.Or;
+            FilterValue.FilterDescriptors = new FilterDescriptorCollection(){
```
Comment (dimodi) [inline components/filter/overview.md:130]:
```suggestion

            FilterValue.FilterDescriptors = new FilterDescriptorCollection(){
```

**Review by yordan-mitev — APPROVED** [review]

**Pull request merged**